src/script/queue.rs

```diff
@@ -0,0 +1,81 @@
+#[derive(Debug)]
```

Comment:
It will just reallocate to push more, its growabl: "VecDeque is a growable ring buffer"